openstudiocore/src/model/ThermalZone.cpp

```diff
@@ -2269,6 +2276,34 @@ namespace detail {
     return result;
   }
 
+  boost::optional<ZoneControlContaminantController> ThermalZone_Impl::zoneControlContaminantController() const
+  {
+    auto h = handle();
+
+    auto controllers = model().getModelObjects<ZoneControlContaminantController>();
+    for( const auto & controller : controllers ) {
+      if( auto zone = controller.getImpl<detail::ZoneControlContaminantController_Impl>()->controlledZone() ) {
+        if( zone->handle() == h ) {
+          return controller;
+        }
+      }
+    }
+
+    return boost::none;
+  }
+
+  bool ThermalZone_Impl::setZoneControlContaminantController(const ZoneControlContaminantController & contaminantController)
+  {
+    auto tz = getObject<ThermalZone>();
+    return contaminantController.getImpl<detail::ZoneControlContaminantController_Impl>()->setControlledZone(tz);
+  }
+
+  void ThermalZone_Impl::resetZoneControlContaminantController()
+  {
+    if( auto controller = zoneControlContaminantController() ) {
+      controller->getImpl<detail::ZoneControlContaminantController_Impl>()->resetControlledZone();
```

Comment:
Otherwise you get orphans that you will never see right?  These things aren't shown in the model list right, just the library list?  I'd go with what you did for thermostats